src/charm_state.py

```diff
@@ -337,8 +381,9 @@ def _parse_dockerhub_mirror(cls, charm: CharmBase) -> str | None:
         if not dockerhub_mirror:
             return None
 
+        dockerhub_mirror = cast(str, dockerhub_mirror)
         dockerhub_mirror_url = urlsplit(dockerhub_mirror)
-        if dockerhub_mirror is not None and dockerhub_mirror_url.scheme != "https":
+        if dockerhub_mirror_url.scheme != "https":
```

Comment:
Out of curiosity, would we be able to leverage HttpUrl with Pydantic and `allowed_schemes` set to `https` for this check?
https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.HttpUrl

(Out of scope for this PR so just leaving a comment, feel free to resolve!)

Reply:
Updated